cdap-common/src/main/java/co/cask/cdap/common/internal/guava/ClassPath.java

```diff
@@ -136,7 +136,10 @@ public static ClassPath from(URI uri, ClassLoader classLoader) throws IOExceptio
    * @since 16.0
    */
   public ImmutableSet<ClassInfo> getAllClasses() {
-    return FluentIterable.from(resources).filter(ClassInfo.class).toImmutableSet();
+    return resources.stream()
+      .filter(ClassInfo.class::isInstance)
+      .map(ClassInfo.class::cast)
+      .collect(Collectors.collectingAndThen(Collectors.toSet(), ImmutableSet::copyOf));
```

Comment:
Use `Collectors.toCollection(LinkedHashSet::new)` instead to preserve the old behavior.

Reply:
Do you mean like this:
```java
.collect(Collectors.collectingAndThen(Collectors.toCollection(LinkedHashSet::new), ImmutableSet::copyOf));
```